subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/EventFiringTaskExecuter.java

```diff
@@ -53,15 +56,25 @@ public TaskExecuterResult call(BuildOperationContext operationContext) {
             }
 
             private TaskExecuterResult executeTask(BuildOperationContext operationContext) {
+                Logger logger = task.getLogger();
+                ContextAwareTaskLogger contextAwareTaskLogger = null;
                 try {
                     taskExecutionListener.beforeExecute(task);
+                    BuildOperationRef currentOperation = buildOperationExecutor.getCurrentOperation();
+                    if (logger instanceof ContextAwareTaskLogger) {
```

Comment:
works as a safety net as we still have TaskInternal#replaceLogger

Reply:
Do we call that internally? Should we decorate the provided logger?

Reply:
I see that `KotlinCompilerWarningSubstitutingLogger` wraps the delegate, so I guess we don't want to decorate it.